### ISSUE TITLE
prefill on already existing cached games

### ISSUE THREAD

**Issue by ich777** [tpill90/steam-lancache-prefill#114]
Hi,
first of all thank you for that wonderful piece of software... :)

I ran into an issue when running the command `./SteamPrefill prefil --recent` on my Debian LXC container that it actually downloads the whole entire game from the Steam servers instead from the `lancache` (because the games where cached before) where on my PC it downloads the game just fine from my `lancache`.

I've also made sure that it can resolve and find the `lancache` in the network.

Here is a screenshot after running the command from above (my Internet is capped at 250Mbit/s down) so it is clearly loading it from their servers and I also noticed that `SteamPrefill` spits a message out something that it downloads it from the CDN but only really short so that I can barely read it before the download starts:
![image](https://user-images.githubusercontent.com/28066518/188118891-52e48930-6112-4585-a70d-72149359d9ca.png)

After a short while I've aborted the download and tried it from my PC where you can see it is clearly downloading it from my `lancache`:
![image](https://user-images.githubusercontent.com/28066518/188119634-b9f0a38b-aced-4d10-8c13-1e1702fbe502.png)

I also checked the access logs where it says `MISS` when running over the `SteamPrefill` (10.0.0.4 is where `SteamPrefill` is on):
![image](https://user-images.githubusercontent.com/28066518/188119849-e7a01558-92d6-4417-bb90-2716f5a241f0.png)

And here you can see that it passes when downloading it from my PC (10.0.0.41 is my PC):
![image](https://user-images.githubusercontent.com/28066518/188119931-94982085-af8d-47ff-b14b-d551e5758a8f.png)

I know that CrossCode is a small game and maybe not the best example.

EDIT: I've also now tried it from my Windows PC, this is when I start `SteamPrefill` (as you can see it's also capped at 250Mbit/s -  so to speak directly from Steam):
![image](https://user-images.githubusercontent.com/28066518/188131952-14bbacee-0a11-4f97-afc6-e8be92f9e4ac.png)

And this is when I download it from the same PC on Steam itself (clearly from `lancache`):
![image](https://user-images.githubusercontent.com/28066518/188132021-97469061-74e5-4c6d-9f0c-796a17774c7e.png)

Am I missing something here or is this just normal behavior?

**Comment by tpill90** [comment]
Hi, thank you for such a detailed report!

No, this is not the expected behavior.  When working as intended **SteamPrefill** should behave exactly like Steam does, in that a cached game will be downloaded only from the cache.

It turns out that several others ran into the exact same issue this morning which was caused by https://github.com/uklans/cache-domains/pull/197 being merged and released two days ago.

I've released https://github.com/tpill90/steam-lancache-prefill/releases/tag/v1.5.0 with the fix for this, which should have everything working as intended.

Since I've had a few other people test and prove the latest release as fixing this issue, I'm going to close it for now.  If for any reason things still aren't working, please feel free to reopen.

**Comment by ich777** [comment]
@tpill90 sorry that I got back late to you but this is not solved, can you please reopen the issue.

It is really frustrating for me when I saw that three games where updated at about the same time (which are about 300GB in size) and they are downloaded entirely from the `lancache` even if the update was only about 2GB in size.

Is this really expected behavior?
For the time being I have sopped using prefill and yes I will have to stop using `prefill` entirely because this consumes more power and slows down my server - using `lancache` on a spinning drive (I/O wait).

**Comment by tpill90** [comment]
Hi @ich77. I just want to clarify a few things. Was this issue ever resolved for you with previous updates?  Or did it start happening again with a newer version?

Also it sounds like you're running prefill in a container, correct?

I will reopen this issue for now until we can figure out what's going on.

**Comment by tpill90** [comment]
Also do you have Discord @ich777 , I would like to talk to you a little bit more in depth on this issue.

**Comment by ich777** [comment]
@tpill90 sorry for the late response, I'm really busy lately, had to fix a few compilation issues for my Driver plugins on Unraid.

Sorry this was maybe the wrong issue to post in since my issue is a little different now. It downloads the already cached parts from the `lancache` as expected but my question would be if there is another way to save or at least a dummy file somehow what it already downloaded and only download the "new" update parts since in my opinion this is a huge waste of CPU cycles if a update from about 2GB is released and it has to pull 110GB from the `lancache` and produces every time the prefill runs huge I/O wait on my system (maybe because I am using a spinning HDD).

Yes, I have Discord too.

**Comment by reg98643t** [comment]
I just updated to the latest version which was working fine and now I am experiencing this issue as well.

**Comment by tpill90** [comment]
@reg98643t I'm going to break out your comment into its own issue, as I think the original issue has already been resolved.  We can continue on the new issue.
